client/me/help/help-contact-form/index.jsx

```diff
@@ -175,6 +197,22 @@ export class HelpContactForm extends React.PureComponent {
 
 	getSibylQuery = () => ( this.state.subject + ' ' + this.state.message ).trim();
 
+	doRequestSite = () => {
+		if ( resemblesUrl( this.state.userDeclaredUrl ) ) {
+			// The API would reject something like "https://wp.com/slug".
+			// It'd need to either be "https://wp.com" or "wp.com".
+			const url = this.state.userDeclaredUrl;
+			const query = url.includes( '://' )
+				? new URL( this.state.userDeclaredUrl ).hostname
+				: new URL( 'http://' + this.state.userDeclaredUrl ).hostname;
+
+			this.props
+				.requestSite( query )
+				.catch( ( error ) => this.setState( { errorData: error.error } ) )
+				.then( ( siteData ) => this.setState( { siteData } ) );
```

Comment:
The order of the handlers should be reversed: `then` first, `catch` second. In the current order, the `then` handler will be called even in case of error, and will set `siteData` to `undefined`. That's the return value of the `catch` callback.

If you really want to reset `siteData` on error, it's better to do so explicitly.

Reply:
Thanks, fixed!